### PR TITLE
Backendmain

### PR DESCRIPTION
**Pull request by realswat1**
Worked on the backend for the main page . At end i tested this implemented routed and changes using postman.

Screenshot 2023-07-12 at 5 21 58 PM
This was a test for the update information for a specific video route .

image
this was a test for the Retrieve information about a specific videos route

image

this  was a test to retrieve a list of all videos using the route

image
  This was a test to create a new video using my route.

Video of a testing
 https://www.loom.com/share/2253229c61bc4d60b4dba0556de1b7bf

### CODE REVIEW

**Comment by alfonsocheng** [comment]
is video the same as stream? in the project plan, the streams table is part of MVP but videos is going to be stretch

image

**Comment by realswat1** [comment]
> is video the same as stream? in the project plan, the streams table is part of MVP but videos is going to be stretch
> 
> image
 
it is streams but not from the subscribed channels.

**Comment by alfonsocheng** [comment]
we chatted offline and decided to keep videos but remove streams

**Comment by alfonsocheng** [comment]
let's make sure the video.js model matches the design in the project plan, or update the project plan to match

File: exposee-app/models/video.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    allowNull: true
+  },
+  duration: { 
+    type: DataTypes.DATE,
+    allowNull:false
+
+  },
+  user_id:{
+    type: DataTypes.INTEGER,
+    allowNull: false,
```
Comment (alfonsocheng) [inline exposee-app/models/video.js:20]: I think you need to do something here to make this a FK

File: exposee-app/models/video.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  },
+  is_saved:{
+    type: DataTypes.BOOLEAN,
+    allowNull: false
+  },
+  updated_at:{
+    type: DataTypes.DATE,
+    allowNull: false
+  },
+  API_key:{
```
Comment (alfonsocheng) [inline exposee-app/models/video.js]: let's stick with all lower case

File: exposee-app/server.js [hunk 3]
```diff
@@ -36,51 +37,70 @@ app.use(
...
-    if (!req.session.user) {
-      return res.status(401).json({ error: 'Unauthorized' });
-    }
+app.get('/videos/:id', async(req,res) => {
 
-    // Retrieve the current user from the session
-    const currentUser = req.session.user;
+  try {
+    const videoId = req.params.id;
+    const videos = await Video.findByPk(videoId)
```
Comment (alfonsocheng) [inline exposee-app/server.js]: so when you use findByPk, it should return 1 video, thus calling this `const video` makes more sense

File: exposee-app/server.js [hunk 4]
```diff
@@ -36,51 +37,70 @@ app.use(
...
+    if (videos){
+  res.json(videos);
+}else  {
+  res.status(404).json({ error:' Video not Found' });
+}
+  }catch (error){
```
Comment (alfonsocheng) [inline exposee-app/server.js:78]: you definitely need to run the eslint with --fix here

**Review by alfonsocheng — APPROVED** [review]

**Comment by alfonsocheng** [comment]
you should update the screenshots if there's a massive change like changing the underlying model but if you tested it already, then it's ok!

File: exposee-app/models/index.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import { User } from './users.js';
-import { Post } from './post.js';
+import { Video } from './video.js';
 
-User.hasMany(Post, { as: 'posts', foreignKey: 'userId' });
-Post.belongsTo(User, { as: 'user', foreignKey: 'userId' });
+User.hasMany(Video, { as: 'videos', foreignKey: 'userId' });
```
Comment (alfonsocheng) [inline exposee-app/models/index.js:4]: I suspect the foreignKey would need to be user_id

File: exposee-app/models/video.js (hunk 1 above)
Comment (alfonsocheng) [reply exposee-app/models/video.js:20]: oh ignore this, I missed the change above in index.js

File: exposee-app/models/video.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  description: {
+    type: DataTypes.STRING,
+    allowNull: true
+  },
+  duration: { 
+    type: DataTypes.DATE,
+    allowNull:false
+
+  },
+  userId:{
```
Comment (alfonsocheng) [inline exposee-app/models/video.js:18]: oh how come this one us userId instead of user_id to match all the other column names?

**Pull request merged**